lib/pure/math.nim

```diff
@@ -211,21 +222,26 @@ func copySign*[T: SomeFloat](x, y: T): T {.inline, since: (1, 5, 1).} =
     doAssert copySign(-Inf, -0.0) == -Inf
     doAssert copySign(NaN, 1.0).isNaN
     doAssert copySign(1.0, copySign(NaN, -1.0)) == -1.0
-
   # TODO: use signbit for examples
-  when defined(js):
-    let uintBuffer = toBitsImpl(y)
-    let sgn = (uintBuffer[1] shr 31) != 0
-    result = jsSetSign(x, sgn)
+
+  template copySignImpl: T =
+    when T is float32:
+      const signMask = 1'u32 shl 31
+      type U = uint32
+    else:
+      const signMask = 1'u64 shl 63
+      type U = uint64
+
+    cast[T]((cast[U](x) and not signMask) or (cast[U](y) and signMask))
```

Comment:
```suggestion
    cast[T]((cast[U](x) and not signMask) or (cast[U](y) and signMask))

```

keeps the template as a logical unit, and the rest of the body of the outer routine with the `when` follows.